src/main/java/com/bobocode/service/impl/AdvisorServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.bobocode.service.impl;
+
+import static com.bobocode.enums.RoleType.ASSOCIATE;
+import static com.bobocode.enums.RoleType.PARTNER;
+import static com.bobocode.enums.RoleType.SENIOR;
+import static com.bobocode.enums.Status.NEW;
+import static java.math.BigDecimal.ZERO;
+import static java.math.BigDecimal.valueOf;
+
+import com.bobocode.domain.Advisor;
+import com.bobocode.domain.Application;
+import com.bobocode.domain.CreditRange;
+import com.bobocode.enums.RoleType;
+import com.bobocode.exception.NoAdvisorFoundException;
+import com.bobocode.exception.NoApplicationFoundException;
+import com.bobocode.repo.AdvisorRepository;
+import com.bobocode.repo.ApplicationRepository;
+import com.bobocode.service.AdvisorService;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AdvisorServiceImpl implements AdvisorService {
+
+  private final ApplicationRepository applicationRepository;
+  private final AdvisorRepository advisorRepository;
+
+  private final Map<RoleType, CreditRange> creditRangeByRoles = Map.of(
+      ASSOCIATE, new CreditRange(ZERO, valueOf(9999)),
+      PARTNER, new CreditRange(valueOf(10000), valueOf(50000)),
+      SENIOR, new CreditRange(valueOf(50001), valueOf(Integer.MAX_VALUE)));
+
+  @Override
+  @Transactional
+  public Application assignApplication(Long advisorId) {
+    log.info("Searching for advisor with id = {}", advisorId);
+    Advisor advisor = advisorRepository.findById(advisorId).orElseThrow(
+        () -> {
+          String msg = "No advisor with id=%s is found".formatted(advisorId);
+          log.error(msg);
+          throw new NoAdvisorFoundException(msg);
+        });
+
+    Application application = findNewApplicationForAdvisor(advisor);
+    advisor.assignApplication(application);
+
+    log.info("Assigned application with id={} to advisor with id={}", application.getId(),
+        advisor.getId());
+
+    return application;
+  }
+
+  private Application findNewApplicationForAdvisor(Advisor advisor) {
+    log.info("Searching for NEW application to assign to the advisor ");
+
+    CreditRange creditRange = getQualifiedCreditRangeByRole(advisor.getRoleType());
+    return applicationRepository.findFirstByStatusAndCreditAmountUSDBetweenOrderByCreatedAt(
+            NEW, creditRange.min(), creditRange.max())
+        .orElseThrow(() -> {
```

Comment:
@kebann again. To me exceptions and logs should be handled carefully, but they should not distract.

Reply:
Agreed that here it makes it hard to read